Makefile

```diff
@@ -68,7 +69,7 @@ clientgen:
 codegen-local: protogen clientgen openapigen manifests-local
 
 .PHONY: codegen
-codegen:
+codegen: dev-tools-image
```

Comment:
please don't make me build the image every time - it should be first time install

Reply:
Docker caching works pretty good when image is built on the same host. Second build is taking no time. Without it developers have to guess if image changed or not.

Reply:
Can you check. I felt like it always rebuilt on my Mac

Reply:
Re-run `dev-tools-image` today morning. 

```
time make dev-tools-image                                                                                                                                                      (minikube/argocd-e2e)
cd hack && docker build -t argocd-dev-tools . -f Dockerfile.dev-tools
Sending build context to Docker daemon  35.84kB
Step 1/10 : FROM golang:1.12.6 as builder
 ---> 9fe4cdc1f173
Step 2/10 : RUN apt-get update && apt-get install -y zip
 ---> Using cache
 ---> e3ab776da8c5
Step 3/10 : ADD install.sh .
 ---> Using cache
 ---> 7032c450906b
Step 4/10 : ADD installers installers
 ---> Using cache
 ---> d847465b0045
Step 5/10 : RUN ./install.sh codegen-tools
 ---> Using cache
 ---> 50a982a96e66
Step 6/10 : RUN ./install.sh codegen-go-tools
 ---> Using cache
 ---> 74d03b80cd5f
Step 7/10 : RUN mkdir -p /home/user && chmod 777 /home/user
 ---> Using cache
 ---> 1907cda1e5e2
Step 8/10 : RUN HELM_HOME=/home/user/.helm helm init --client-only
 ---> Using cache
 ---> b53f296a6ac7
Step 9/10 : RUN mkdir -p /go/pkg && chmod 777 /go/pkg
 ---> Using cache
 ---> 5990a9696ea7
Step 10/10 : RUN apt-get clean &&     rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 ---> Using cache
 ---> ec59cf0160ff
Successfully built ec59cf0160ff
Successfully tagged argocd-dev-tools:latest
make dev-tools-image  0.11s user 0.10s system 35% cpu 0.580 total
```